test/integration/autolinker_spec.mjs

```diff
@@ -121,4 +121,42 @@ describe("autolinker", function () {
       );
     });
   });
+
+  describe("PR 19470", function () {
+    let pages;
+
+    beforeAll(async () => {
+      pages = await loadAndWait(
+        "bug1019475_2.pdf",
+        ".annotationLayer",
+        null,
+        null,
+        {
+          enableAutoLinking: true,
+        }
+      );
+    });
+
+    afterAll(async () => {
+      await closePages(pages);
+    });
+
+    it("must not repeatedly add link annotations redundantly", async () => {
+      await Promise.all(
+        pages.map(async ([browserName, page]) => {
+          let url = await page.$$eval(
+            ".annotationLayer > .linkAnnotation > a",
+            annotations => annotations.map(a => a.href)
+          );
+          expect(url.length).withContext(`In ${browserName}`).toEqual(1);
+          await page.evaluate(() => window.PDFViewerApplication.zoomReset());
```

Comment:
I believe that there's a couple of issues with this line:
 - It *implicitly* relies on the initial zoom level of the document being "page-fit", and if the default zoom-value for the integration-tests is ever changed this test would no longer be testing the intended thing, so it'd obviously be much better to invoke zoomIn or zoomOut here.
 - Just awaiting the zoom-call probably doesn't really guarantee that re-rendering has finished *or* that the annotationLayer has been re-enabled (since it's hidden during zooming). Hence you need to wait for the appropriate events, since otherwise the checks below might not be testing the correct thing.
 - Lastly, having a blank-line before this "second part" of the test-case wouldn't hurt.

Reply:
@ryzokuken You can take a look at this test for how to wait for the zoom to happen: https://github.com/mozilla/pdf.js/blob/d6f63d0e4b35b31355bdf4f8338e284bf72c75b2/test/integration/viewer_spec.mjs#L231

Reply:
Please also see PR #19475 which adds a new event that might be helpful here, since it'll allow detecting when the inferred-links have actually been added.

Reply:
Great, thanks for your PR, will rebase on top once it's merged and listen on that event to avoid zooming in before the links are injected.

Reply:
I believe all three of your points were addressed, so I'm marking this as resolved.